features/admin.roles.v2/components/wizard-updated/role-basics.tsx

```diff
@@ -208,14 +177,54 @@ export const RoleBasics: FunctionComponent<RoleBasicProps> = (props: RoleBasicPr
                 ? values.assignedApplicationId.toString()
                 : null,
             assignedApplicationName: values.roleAudience === RoleAudienceTypes.APPLICATION
-                ? applicationListOptions?.find((application: DropdownProps) =>
-                    application.key === values.assignedApplicationId.toString())?.text
+                ? filteredApplicationList?.find((application: ApplicationListItemInterface) =>
+                    application.id === values.assignedApplicationId.toString())?.name
                 : null,
             roleAudience: values.roleAudience.toString(),
             roleName: values.roleName.toString()
         };
     };
 
+    /**
+     * Generates the dropdown options list for the available applications.
+     *
+     * @param _applicationList - list of applications to be displayed.
+     * @returns The dropdown options list for the applications.
+     */
+    const getAvailableApplicationOptions = (
+        _applicationList: ApplicationListItemInterface[]
+    ): DropdownProps[] => {
```

Comment:
Do we handle application filtering as well in the application selection dropdown?